ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_dev/ExplicitDisclosureIT.scala

```diff
@@ -0,0 +1,417 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.suites.v1_dev
+
+import com.daml.error.definitions.LedgerApiErrors
+import com.daml.ledger.api.testtool.infrastructure.Allocation._
+import com.daml.ledger.api.testtool.infrastructure.Assertions._
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.testtool.infrastructure.TransactionHelpers.createdEvents
+import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
+import com.daml.ledger.api.v1.command_service.SubmitAndWaitRequest
+import com.daml.ledger.api.v1.commands.{Command, DisclosedContract, ExerciseByKeyCommand}
+import com.daml.ledger.api.v1.event.CreatedEvent
+import com.daml.ledger.api.v1.transaction_service.GetTransactionsRequest
+import com.daml.ledger.api.v1.value.{Record, RecordField, Value}
+import com.daml.ledger.client.binding
+import com.daml.ledger.client.binding.Primitive
+import com.daml.ledger.test.model.Test._
+import com.daml.lf.value.Value.ContractId
+import scalaz.syntax.tag._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+// TODO ED: Enable in Sandbox-on-X conformance test-suite
+final class ExplicitDisclosureIT extends LedgerTestSuite {
+  import ExplicitDisclosureIT._
+
+  test(
+    "EDCorrectDisclosure",
+    "Submission works if the correct disclosure is provided",
+    allocate(Parties(2)),
+    enabled = _.explicitDisclosure,
+  )(implicit ec => { case Participants(Participant(ledger, owner, delegate)) =>
+    for {
+      testContext <- initializeTest(ledger, owner, delegate)
+
+      // Exercise a choice on the Delegation that fetches the Delegated contract
+      // Fails because the submitter doesn't see the contract being fetched
+      exerciseFetchError <- testContext.exerciseFetchDelegated().failed
+
+      // Exercise the same choice, this time using correct explicit disclosure
+      _ <- testContext.exerciseFetchDelegated(testContext.disclosedContract)
+
+      // Create an extra disclosed contract
+      extraKey = ledger.nextKeyId()
+      _ <- ledger.create(owner, Delegated(owner, extraKey))
+      delegatedTxs <- ledger.flatTransactionsByTemplateId(Delegated.id, owner)
+      extraDelegatedEvent = createdEvents(delegatedTxs(1)).head
```

Comment:
why it's a second element`delegatedTxs(1)` actually? what happens wit the second?

Reply:
The first one is extracted as part of the `testContext` (see `initializeTest`) and used on the first `FetchDelegated` (see line 45). The second one is used as a superfluous provided disclosed contract (used to assert that non-necessary provided disclosed contracts do not crash a submission)

Reply:
I'm not following. The `extraDelegatedEvent` is an actual transaction create event (as part of the second transaction). Whereas `ledger.create` returns a `ContractId` and not the actual contract payload.

Reply:
Sorry my bad, I expressed myself very poorly. I'm not asking for any change. I just got slightly confused when reading this because I thought the `1` was an index in the evens of the transaction created by the `create` rather than an index into all transactions for the template.